File: common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Item.scala
package uk.ac.wellcome.models.work.internal

case class Item[+State](
  id: State,
  title: Option[String] = None,
  locations: List[LocationDeprecated] = Nil
) extends HasId[State]

object Item {

  def apply[State >: IdState.Unidentifiable.type](
    title: Option[String],
    locations: List[LocationDeprecated]
  ): Item[State] =
    Item(IdState.Unidentifiable, title, locations)
}

File: catalogue_graph/src/models/pipeline/item.py
from models.pipeline.identifier import Identifiable, Identified, Unidentifiable
from models.pipeline.location import DigitalLocation, PhysicalLocation
from models.pipeline.serialisable import SerialisableModel


class Item(SerialisableModel):
    # `Identifiable` covers the source stage, before the id-minter has assigned a
    # canonical id (e.g. FOLIO items carrying a `folio-item` source identifier);
    # `Identified` is the post-mint state; `Unidentifiable` is for items with no
    # stable source identifier.
    id: Identified | Unidentifiable | Identifiable
    title: str | None = None
    note: str | None = None
    locations: list[PhysicalLocation | DigitalLocation] = []
